### PR TITLE
Explanation of behaviour when block:false

### PR DESCRIPTION
**Pull request by orbifx**

### CODE REVIEW

File: src/ZMQ.mli [hunk 1]
```diff
@@ -55,7 +55,9 @@ module Socket : sig
   val unbind : 'a t -> string -> unit
 
   (** Read a message from the socket.
-      block indicates if the call should be blocking or non-blocking. Default true
+      block indicates if the call should be blocking or non-blocking.
+      If block:false, on lack of message to receive, recv raises Unix.Unix_error (EAGAIN, "zmq_msg_recv", "")
```
Comment (hcarty) [inline src/ZMQ.mli]: Suggestion:
```
If block is [false], [recv] will raise [Unix.Unix_error (Unix.EAGAIN, _, _)] if there are no messages available to receive on the specified socket.
```

**Comment by hcarty** [comment]
It may be a good idea in general to specify that the dontwait/non-blocking functionality is intended for use with libraries like Lwt/Async (lwt-zmq and async-zmq).

**Comment by orbifx** [comment]
Most likely. We can't introduce Lwt in our case just now, we are actually using this with OCaml threads.

**Comment by hcarty** [comment]
@orbifx Very nice - any chance you could contribute an example?

**Comment by orbifx** [comment]
Sure. Of what exactly?

**Comment by orbifx** [comment]
Will I add that example in a different merge?

**Comment by hcarty** [comment]
I think it makes sense as a separate PR.  Thank you!

**Comment by orbifx** [comment]
What about this PR?

**Pull request merged**